model/link_metadata.go

```diff
@@ -21,6 +23,46 @@ const (
 
 type LinkMetadataType string
 
+// checks if original string has more than 300 characters long, if so
+// it'll truncate it and add an ellipsis (it will be stripped within
+// the client, but in case it isn't we might as well add it so it is
+// properly understood)
+func truncateText(original string) string {
+	if utf8.RuneCountInString(original) > 300 {
+		return fmt.Sprintf("%.300s[...]", original)
+	}
+	return original
+}
+
+func firstImage(images []*opengraph.Image) []*opengraph.Image {
+	if len(images) > 1 {
+		return []*opengraph.Image{images[0]}
+	}
+	return images
+}
+
+// TruncateOpenGraph modifies a OG into a smaller version to ensure it
+// doesn't grow too big, as that much text won't be displayed by the
+// clients anyway. Also remove unwanted fields
+func TruncateOpenGraph(ogdata *opengraph.OpenGraph) *opengraph.OpenGraph {
+	if ogdata != nil {
+		// we might want to truncate url too, but that can have unintended effect
+		ogdata.Title = truncateText(ogdata.Title)
+		ogdata.Description = truncateText(ogdata.Description)
+		ogdata.SiteName = truncateText(ogdata.SiteName)
+		ogdata.Article = nil
+		ogdata.Book = nil
+		ogdata.Profile = nil
+		ogdata.Determiner = ""
+		ogdata.Locale = ""
+		ogdata.LocalesAlternate = make([]string, 0)
+		ogdata.Images = firstImage(ogdata.Images)
```

Comment:
ah! didn't know about it, I'll review that code. For some reason I thought that "first one" was the option to look for.

Reply:
one thing to consider is how does this image affect mobile, a device can have different sizes as shown here: https://github.com/mattermost/mattermost-mobile/blob/master/app/components/post_attachment_opengraph/post_attachment_opengraph.js#L161
is there someone who can help with this? do we consider webapp the master choice for this?

Reply:
@hmhealey @enahum can you help jesus and me with this? I don't have a strong preference for any format, but you'll have better insight!

Reply:
Oh, I didn't realize that we can use multiple images if those are specified. I figured we just used the first one that was available. Since we pick the image based on screen size, I don't think the server can reliably choose one image to keep.

Reply:
And what do you think? Pick the first 10? Pick them in a smart way based on a set of predefined sizes based on the most common mobile resolutions and on the desktop constant used for this?

Reply:
Honestly, I'm still leaning towards just removing the max length from the column instead of trying to guess which images to keep here. Since we're just storing arbitrary json in a column that doesn't need a search index, I feel like I made a mistake in the first place by limiting the length of the data.

I think it's hard to figure out what size of image a client wants since it depends on what that specific client wants to just show. It looks like the web app tries to use the smallest image possible for some reason, while the mobile app tries to find the largest one that fits its screen, and we may decide to change either of those in the future.

Reply:
I still think this solution makes more sense. I would suggest to go for a simple solution here where we accept the first 15 images (I don't think more makes any sense), and then increase the size of the field. Actually, I would prefer to get this merged because is isolated enough, and then add another PR removing the size limit for the field. @hmhealey thoughs? @Willyfrog thoughs?

Reply:
Hi @Willyfrog, Jesús and I chatted for a bit to discuss this, and we decided that 5 images should be enough to show the preview correctly. We'd still like to remove the size limit for the field (which can be done in a separate PR), but keeping it at 5 images will at least save a bit of space and bandwidth.

Let us know if you have any questions. I understand that you may be busy with other stuff right now though, so feel free to take your time.

Reply:
that should be easy to implement.

Sorry for the big delay, but life got in the way and was a bit difficult to keep up with everything.